cpp/src/decisiontree/levelalgo/levelfunc_classifier.cuh

```diff
@@ -88,6 +88,19 @@ void grow_deep_tree_classification(
   unsigned int* h_new_node_flags = tempmem->h_new_node_flags->data();
   unsigned int* d_new_node_flags = tempmem->d_new_node_flags->data();
   unsigned int* d_colids = tempmem->d_colids->data();
+  unsigned int* h_colids = tempmem->h_colids->data();
+  unsigned int* d_colstart = nullptr;
+  unsigned int* h_colstart = nullptr;
```

Comment:
I think this is just my limited understanding, but why do we need to track colstart and colids on the host in addition to device?

Reply:
Those ones are needed for populating the right featureid in the Decision Tree data structure.